File: chat-uikit/src/main/java/io/agora/chat/uikit/menu/EaseChatType.java
package io.agora.chat.uikit.menu;

public enum EaseChatType {
    UNKNOWN(0),
    SINGLE_CHAT(1),
    GROUP_CHAT(2),
    CHATROOM(3);

    private int chatType;

    private EaseChatType(int chatType) {
        this.chatType = chatType;
    }

    public int getChatType() {
        return chatType;
    }

    public static EaseChatType from(int chatType) {
        for (EaseChatType type: EaseChatType.values()) {
            if(type.chatType == chatType) {
                return type;
            }
        }
        return UNKNOWN;
    }
}

File: ease-im-kit/src/main/kotlin/com/hyphenate/easeui/feature/chat/enums/EaseChatType.kt
package com.hyphenate.easeui.feature.chat.enums

import com.hyphenate.easeui.common.ChatConversationType

enum class EaseChatType {
    /**
     * Single chat type.
     */
    SINGLE_CHAT,

    /**
     * Group chat type.
     */
    GROUP_CHAT,

    /**
     * Chat room type.
     */
    CHATROOM
}

fun EaseChatType.getConversationType(): ChatConversationType {
    return when (this) {
        EaseChatType.GROUP_CHAT -> ChatConversationType.GroupChat
        EaseChatType.CHATROOM -> ChatConversationType.ChatRoom
        else -> ChatConversationType.Chat
    }
}